Utilities/rANS/include/rANS/internal/helper.h

```diff
@@ -142,6 +145,15 @@ class JSONArrayLogger
   bool mReverse{false};
 };
 
+inline uint32_t safeadd(uint32_t a, uint32_t b)
+{
+  uint32_t result;
+  if (rans_unlikely(__builtin_uadd_overflow(a, b, &result))) {
```

Comment:
is this portable?

Reply:
Yes. I checked for all compilers we support (GCC>7, CLANG), however they might perform differently between compilers and architectures (like arm vs x86). The reason why I am using this, is because compilers have low cost access to CPU flags that I cannot access in a portable way (like the carry bit that is set when overflow occurs).